stdweb-derive/src/lib.rs

```diff
@@ -165,14 +178,25 @@ pub fn derive_reference_type( input: TokenStream ) -> TokenStream {
         }
     }
     let default_args = quote! { #(#default_args),* };
+
     let instance_of_impl = match instance_of {
         Some( js_name ) => {
+            let mut js_constraints : Vec<String> = constraints.into_iter().map(|constraint|{
+                let js_field = constraint.0.replace("\"", "\\\"");
+                let js_value = constraint.1.replace("\"", "\\\"");
+                format!("_ref[\"{}\"] === \"{}\"", &js_field, &js_value)
+            }).collect();
+
+            js_constraints.insert( 0, format!("_ref instanceof {}", &js_name) );
+
+            let js_constraints = js_constraints.join(" && ");
+
```

Comment:
Could we have semi-separate codepaths for this? (That is - replace the whole body of `instance_of` with `#instance_of_body`, have the old case of having only `instance_of` generate code exactly as it used to, and only generate a differente `instance_of_body` in case of a constraint.)